libsolidity/interface/Natspec.cpp

```diff
@@ -255,3 +255,37 @@ Json::Value Natspec::devDocumentation(std::multimap<std::string, DocTag> const&
 
 	return json;
 }
+
+vector<EventDefinition const*>  Natspec::uniqueInterfaceEvents(ContractDefinition const& _contract)
+{
+	auto eventSignature = [](EventDefinition const* _event) -> string {
+		FunctionType const* functionType = _event->functionType(true);
+		solAssert(functionType, "");
+		return functionType->externalSignature();
+	};
+	auto compareBySignature =
+		[&](EventDefinition const* _lhs, EventDefinition const* _rhs) -> bool {
+			return eventSignature(_lhs) < eventSignature(_rhs);
+		};
+
+	set<EventDefinition const*, decltype(compareBySignature)> uniqueEvents{compareBySignature};
+	// Insert events defined in the contract first so that in case of a conflict
+	// they're the ones that get selected.
+	uniqueEvents += _contract.definedInterfaceEvents();
+
+	set<EventDefinition const*, decltype(compareBySignature)> filteredUsedEvents{compareBySignature};
+	set<string> usedSignatures;
+	pair<set<string>::iterator, bool> signatureResult;
+	pair<set<EventDefinition const*, decltype(compareBySignature)>::iterator, bool> eventResult;
+	for (EventDefinition const* event: _contract.usedInterfaceEvents())
+	{
+		eventResult = filteredUsedEvents.insert(event);
+		signatureResult = usedSignatures.insert(eventSignature(event));
+		if (signatureResult.second == false)
+				// Either the old event or the new one will be erased
+				filteredUsedEvents.erase(eventResult.first);
+	}
```

Comment:
```suggestion
	for (EventDefinition const* event: _contract.usedInterfaceEvents())
	{
		auto&& [event, eventInserted] = filteredUsedEvents.insert(event);
		auto&& [signature, signatureInserted] = usedSignatures.insert(eventSignature(event));
		if (!signatureInserted)
				// Either the old event or the new one will be erased
				filteredUsedEvents.erase(eventIt);
	}
```

Reply:
> // Either the old event or the new one will be erased

By the way, it seems to me that it will always be the new one. Unless `usedInterfaceEvents()` can return the same event multiple times, but I don't think that's the case (easy to find out by asserting that `eventInserted` is `true`).

Reply:
Thanks for the suggestion of the simplification!

> By the way, it seems to me that it will always be the new one. 

I think that only in the first "collision" it will be the old one and then always the new.
In this example, here L1.E will be added first, then L2.E would not be inserted and L1.E erased. Then, L3.E is erased and any other new one with the same signature:
```solidity
library L1 { event E(uint); }
library L2 { event E(uint); }
library L3 { event E(uint); }
contract C { 
    function f() public {
        emit L1.E(1);
        emit L2.E(1);
        emit L3.E(1);
    }
}
```
I was going to add a `L3` to the test case with different libraries because I thought of this case when implementing this. What do you think? overkill?

Reply:
> I think that only in the first "collision" it will be the old one and then always the new.

Ah, true. I missed the fact that `filteredUsedEvents` is ordered by signature. So it will reject not only identical events but also events with same signature.

> I was going to add a L3 to the test case with different libraries because I thought of this case when implementing this. What do you think? overkill I guess, right?

I think it's a good idea. Adding an extra event to an existing case does not add that much code. It's adding new cases that's annoyingly verbose :)